processor/transformprocessor/internal/traces/registry.go

```diff
@@ -0,0 +1,84 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+var registry = make(map[string]interface{})
```

Comment:
See previous comments, not sure this is useful at this moment, we should accept the "AvailableFuncs" everywhere, and if needed only the top level call can offer the "registry" pattern. At least we make all the other funcs work with a provided funcs.

Reply:
One reason I went with a global was that we can easily imagine multiple instances of the processor for different pipelines. Just want to confirm it's OK to have registry's with the same content initialized for each of them? Or let me know if I'm misunderstanding

Reply:
Indeed, but all the instances for the processor are created via the given ProcessorFactory instance, so will "share" all available funcs without having to have this registry here.

Reply:
I tried changing the pattern but noticed that it would prevent me from using `UnmarshalText`. It resolves functions in query, to allow the config validation to return an error immediately if an unknown function is referenced

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7300/files#diff-52802c1c43b30e51469dffde6bfa9ccd59c5d4778e641f0dd983036b20d865c0R75

Because it has no reference to a processor or factory, I seem to only be able to reference the global. I think `Config.Validate()` would have the same problem. I think to have injected function registry, I would have to go back to deferring config validation to processor initialization which you pointed out is not a good pattern. Am I missing an alternative, or is the global OK given this limitation?

Reply:
You can do a "Config.Validate()" for sure, since the Config instance is created by the `Factory` via the "CreateDefaultConfig". 

The  `UnmarshalText` is a good point, but is this an "Unmarshal" error or a "Validation" error?

Reply:
Do you mean pass state from the factory to the Validate() function using a private member like this? I think it'll work but it seems a bit mysterious passing state like that, but it seems OK too.

```
func createDefaultConfig() config.Processor {
	return &Config{
		ProcessorSettings: config.NewProcessorSettings(config.NewComponentID(typeStr)),
		Traces: TracesConfig{
			Queries: []string,
            funcs: map[string]interface{}
		},
	}
}
```

Reply:
Kind of like that

```golang
func createDefaultConfig() config.Processor {
	return &Config{
		ProcessorSettings: config.NewProcessorSettings(config.NewComponentID(typeStr)),
		Traces: TracesConfig{
			Queries: []traces.Query,
			funcs: map[string]interface{} `mapstructure:"-"` // to not be able to set this.
		},
	}
}

func (cfg *Config) Validate() {
  return multierr.Combine(cfg.Traces.Validate(), cfg.Metrics.Validate(), ...)
}

func (tc *TracesConfig) Validate() {
   ...
}
```